src/wallet/wallet.cpp

```diff
@@ -274,7 +274,7 @@ bool CWallet::ChangeWalletPassphrase(const SecureString& strOldWalletPassphrase,
                     return false;
                 if (!crypter.Encrypt(vMasterKey, pMasterKey.second.vchCryptedKey))
                     return false;
-                CWalletDB(strWalletFile).WriteMasterKey(pMasterKey.first, pMasterKey.second);
```

Comment:
There is a reason we create CWalletDB objects every time per operation. From what I remember this assures that the transactions are closed/flushed properly.